Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidExcludeFromCodeCovrageAnalyzer.cs

```diff
@@ -0,0 +1,38 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class AvoidExcludeFromCodeCoverageAnalyzer : SingleDiagnosticAnalyzer<AttributeSyntax, AvoidExcludeFromCodeCoverageAnalyzerSyntaxNodeAction>
+	{
+		private const string Title = @"Avoid the ExcludeFromCodeCoverage attribute";
+		public const string MessageFormat = Title;
+		private const string Description = MessageFormat;
+
+		public AvoidExcludeFromCodeCoverageAnalyzer()
+			: base(DiagnosticId.AvoidExcludeFromCodeCoverage, Title, MessageFormat, Description, Categories.Maintainability)
+		{ }
+	}
+
+	public class AvoidExcludeFromCodeCoverageAnalyzerSyntaxNodeAction : SyntaxNodeAction<AttributeSyntax>
+	{
+		private const string ExcludeFromCodeCoverageAttributeTypeName = "ExcludeFromCodeCoverage";
+		public override void Analyze()
+		{
+			IReadOnlyDictionary<string, string> aliases = Helper.ForNamespaces.GetUsingAliases(Node);
+			if (!Node.Name.GetFullName(aliases).Contains(ExcludeFromCodeCoverageAttributeTypeName))
```

Comment:
Because there's only 1 fail-fast clause, suggest inverting (i.e., remove negation and embedded return)

Reply:
ReSharper suggested the opposite ;-)

Reply:
But I agree with you, more readable to remove the negation, did that